.ibm/pipelines/openshift-tests.sh

```diff
@@ -153,13 +153,34 @@ run_tests() {
 }
 
 check_backstage_running() {
-  # Check if Backstage is up and running
-  BACKSTAGE_URL_RESPONSE=$(curl -Is "https://${RELEASE_NAME}-${NAME_SPACE}.${K8S_CLUSTER_ROUTER_BASE}" | head -n 1)
+  local url="https://${RELEASE_NAME}-backstage-${NAME_SPACE}.${K8S_CLUSTER_ROUTER_BASE}"
+  
+  # Maximum number of attempts to check URL
+  local max_attempts=30
+  # Time in seconds to wait
+  local wait_seconds=30
+
+  echo "Checking if Backstage is up and running at $url"
+
+  for ((i=1; i<=max_attempts; i++)); do
+    # Get the status code
+    local http_status=$(curl -o /dev/null -s -w "%{http_code}\n" "$url")
+
+    # Check if the status code is 200
+    if [[ $http_status -eq 200 ]]; then
+      echo "Backstage is up and running!"
+      export BASE_URL=$url
+      echo "######## BASE URL ########"
+      echo "$BASE_URL"
+      return 0
+    else
+      echo "Attempt $i of $max_attempts: Backstage not yet available (HTTP Status: $http_status)"
+      sleep $wait_seconds
+    fi
+  done
 
-  echo "$BACKSTAGE_URL_RESPONSE"
-  export BASE_URL="https://${RELEASE_NAME}-${NAME_SPACE}.${K8S_CLUSTER_ROUTER_BASE}"
-  echo "######## BASE URL ########"
-  echo "$BASE_URL"
+  echo "Failed to reach Backstage after $max_attempts attempts."
```

Comment:
why not echo the URL that failed for easier debugging? 

```suggestion
  echo "Failed to reach Backstage at $BASE_URL after $max_attempts attempts."
```